text/0000-mergeable-rustdoc-cross-crate-info.md

```diff
@@ -0,0 +1,439 @@
+│               │           └── trait.UnwindSafe.js
+│               └── t
+│                   └── trait.T.js
+├── merged
+│   └── doc
+│       ├── crates.js
+│       ├── help.html
+│       ├── i
+│       │   ├── all.html
+│       │   ├── index.html
+│       │   └── sidebar-items.js
+│       ├── index.html
+│       ├── s
+│       │   ├── all.html
+│       │   ├── index.html
+│       │   ├── sidebar-items.js
+│       │   └── struct.S.html
+│       ├── search.desc
+│       │   ├── i
+│       │   │   └── i-desc-0-.js
+│       │   ├── s
+│       │   │   └── s-desc-0-.js
+│       │   └── t
+│       │       └── t-desc-0-.js
+│       ├── search-index.js
+│       ├── settings.html
+│       ├── src
+│       │   ├── i
+│       │   │   └── lib.rs.html
+│       │   ├── s
+│       │   │   └── lib.rs.html
+│       │   └── t
+│       │       └── lib.rs.html
+│       ├── src-files.js
+│       ├── static.files
+│           │   ├── COPYRIGHT-23e9bde6c69aea69.txt
+│           │   ├── favicon-2c020d218678b618.svg
+│           │   └── <rest of the contents excluded>
+│       ├── t
+│       │   ├── all.html
+│       │   ├── index.html
+│       │   ├── sidebar-items.js
+│       │   └── trait.T.html
+│       └── trait.impl
+│           ├── core
+│           │   ├── marker
+│           │   │   ├── trait.Freeze.js
+│           │   │   ├── trait.Send.js
+│           │   │   ├── trait.Sync.js
+│           │   │   └── trait.Unpin.js
+│           │   └── panic
+│           │       └── unwind_safe
+│           │           ├── trait.RefUnwindSafe.js
+│           │           └── trait.UnwindSafe.js
+│           └── t
+│               └── trait.T.js
+├── s
+│   ├── src
+│   │   └── lib.rs
+│   └── target
+│       ├── doc
+│       │   ├── help.html
+│       │   ├── .lock
+│       │   ├── s
+│       │   │   ├── all.html
+│       │   │   ├── index.html
+│       │   │   ├── sidebar-items.js
+│       │   │   └── struct.S.html
+│       │   ├── search.desc
+│       │   │   └── s
+│       │   │       └── s-desc-0-.js
+│       │   ├── settings.html
+│       │   └── src
+│       │       └── s
+│       │           └── lib.rs.html
+│       ├── doc.parts
+│       │   └── s
+│       │       └── crate-info.json
+│       └── libs.rmeta
+└── t
+    ├── src
+    │   └── lib.rs
+    └── target
+        ├── doc
+        │   ├── help.html
+        │   ├── .lock
+        │   ├── search.desc
+        │   │   └── t
+        │   │       └── t-desc-0-.js
+        │   ├── settings.html
+        │   ├── src
+        │   │   └── t
+        │   │       └── lib.rs.html
+        │   └── t
+        │       ├── all.html
+        │       ├── index.html
+        │       ├── sidebar-items.js
+        │       └── trait.T.html
+        ├── doc.parts
+        │   └── t
+        │       └── crate-info.json
+        └── libt.rmeta
+</pre>
+
+</details>
+
+## Suggested workflows
+
+### No cross-crate information 
+Provide `--merge=none` to every invocation of rustdoc.
```

Comment:
Now that I'm thinking about it: couldn't it be the default so there is no need to provide it all the time? Therefore we could remove the `none` value and just expect that if `--merge` isn't provided, then this option is disabled.

Reply:
`--merge=read-write` should be implied if no `--merge` option is provided, because `--merge=read-write` is designed to match the existing behavior of rustdoc. Cargo, the rustdoc test suite, and other tools that use rustdoc expect to be able to run rustdoc with a shared output directory, and get cross-crate information. If `--merge=none` was made the default, these tools would lose cross-crate info, unless they are patched to use `--merge=read-write`.

Reply:
Arf, sorry yes you're right. Please ignore this comment. ^^'